### ISSUE TITLE
[Known Issues] - TensorRT

### ISSUE THREAD

**Issue by NevermindNilas** [NevermindNilas/TheAnimeScripter#24]
Rife has seen an up to 4x performance loss with TensorRT 9.3 and above as per [Amusement Club's](https://github.com/AmusementClub/vs-mlrt/releases/tag/v14#:~:text=Accoding%20to%20the%20documentation%2C%20There%20is%20an%20up%20to%204x%20performance%20regression%20for%20networks%20containing%20%22GridSample%22%20ops%20compared%20to%20TensorRT%209.2.This%20affects%20RIFE%20and%20SAFA%20models.) reports.

Rife TensorRT has some aliasing issues with outputs, I am looking into it.

**Comment by VicMahoney** [comment]
> Rife has seen an up to 4x performance loss with TensorRT 9.3 and above as per [Amusement Club's](https://github.com/AmusementClub/vs-mlrt/releases/tag/v14#:~:text=Accoding%20to%20the%20documentation%2C%20There%20is%20an%20up%20to%204x%20performance%20regression%20for%20networks%20containing%20%22GridSample%22%20ops%20compared%20to%20TensorRT%209.2.This%20affects%20RIFE%20and%20SAFA%20models.) reports.
> 
> Rife TensorRT has some aliasing issues with outputs, I am looking into it.

Hello! Will the gui be fixed so that non-professionals in Python (like me) can use the program?

**Comment by NevermindNilas** [comment]
Hi,

Eventually yes, it's just a pain in the ass to work with and I have been procrastinating it. In the meantime using the [PARAMETERS](https://github.com/NevermindNilas/TheAnimeScripter/blob/main/PARAMETERS.MD) and the main.exe would be the only viable solution and requires no real programming knowledge.



> > Rife has seen an up to 4x performance loss with TensorRT 9.3 and above as per [Amusement Club's](https://github.com/AmusementClub/vs-mlrt/releases/tag/v14#:~:text=Accoding%20to%20the%20documentation%2C%20There%20is%20an%20up%20to%204x%20performance%20regression%20for%20networks%20containing%20%22GridSample%22%20ops%20compared%20to%20TensorRT%209.2.This%20affects%20RIFE%20and%20SAFA%20models.) reports.
> > Rife TensorRT has some aliasing issues with outputs, I am looking into it.
> 
> Hello! Will the gui be fixed so that non-professionals in Python (like me) can use the program?

**Comment by VicMahoney** [comment]
> Hi,
> 
> Eventually yes, it's just a pain in the ass to work with and I have been procrastinating it. In the meantime using the [PARAMETERS](https://github.com/NevermindNilas/TheAnimeScripter/blob/main/PARAMETERS.MD) and the main.exe would be the only viable solution and requires no real programming knowledge.
> 
> > > Rife has seen an up to 4x performance loss with TensorRT 9.3 and above as per [Amusement Club's](https://github.com/AmusementClub/vs-mlrt/releases/tag/v14#:~:text=Accoding%20to%20the%20documentation%2C%20There%20is%20an%20up%20to%204x%20performance%20regression%20for%20networks%20containing%20%22GridSample%22%20ops%20compared%20to%20TensorRT%209.2.This%20affects%20RIFE%20and%20SAFA%20models.) reports.
> > > Rife TensorRT has some aliasing issues with outputs, I am looking into it.
> > 
> > 
> > Hello! Will the gui be fixed so that non-professionals in Python (like me) can use the program?

so its working now?i cant run (cant understand what to do without gui)

**Comment by NevermindNilas** [comment]
No, the GUI.exe is not functional.

**Comment by VicMahoney** [comment]
> No, the GUI.exe is not functional.

so i mean without gui is that real to try this app?

**Comment by NevermindNilas** [comment]
I don't think I understand your question. Can you rephrase it.

**Comment by VicMahoney** [comment]
Is it possible, for example, to interpolate a video through a terminal command without using the GUI shell?

**Comment by NevermindNilas** [comment]
As mentioned previously, yes, it is possible through the use of the main.exe. 

Here are all the available commands that the terminal accepts [PARAMETERS](https://github.com/NevermindNilas/TheAnimeScripter/blob/main/PARAMETERS.MD)

**Comment by VicMahoney** [comment]
please tell me the command to run it
i just enter the folder, and what then?

**Comment by VicMahoney** [comment]
i need to download this as zip from main page or from download menus?

**Comment by NevermindNilas** [comment]
If you want a plug and play experience without the need to install python and alongside it the dependencies, then you download from the release page.

In the .7z file you will find a main.exe file that is basically the brains of everything.

I Would strongly advise getting familiar with the [Paramaters](https://github.com/NevermindNilas/TheAnimeScripter/blob/main/PARAMETERS.MD) if you want to use the script to its full potential.

At the bottom of that very page you have a couple of examples in which you just replace the `python main.py` part with `main.exe` followed by the commands you wish to use.

**Comment by NevermindNilas** [comment]
Here's an example of how to interpolate:
`main.exe --input {path_here} --interpolate 1 --interpolate_method rife4.15`

**Comment by VicMahoney** [comment]
> Here's an example of how to interpolate: `main.exe --input {path_here} --interpolate 1 --interpolate_method rife4.15`

its ok now,thanks
testing

**Comment by VicMahoney** [comment]
but I need a gui to put a lot of videos on the production stream)

**Comment by NevermindNilas** [comment]
That's great to hear.

**Comment by VicMahoney** [comment]
> That's great to hear.

how to interpolate factor 2,5 ?

**Comment by NevermindNilas** [comment]
There's no option for that.

**Comment by NevermindNilas** [comment]
> but I need a gui to put a lot of videos on the production stream)

I haven't tested it much but technically you could also do `--input {folder_path}` and the script should recognize that the input is a folder and process all videos in that select folder.

**Comment by VicMahoney** [comment]
> There's no option for that.

so sad, there are many 23fps videos and to do 60 i am usually set x2.5 factor in flowframes

**Comment by NevermindNilas** [comment]
Maybe [Enhancr](https://github.com/mafiosnik777/enhancr) has support for it but I am not sure. To my knowledge it is not something trivial to implement through the way TAS handles interpolation and I am not planning to look into it for the foreseeable future. For now stability and performance are the main focus.

**Comment by NevermindNilas** [comment]
Fixed with https://github.com/NevermindNilas/TheAnimeScripter/commit/22ed7a7d239323427efcfec43b776c6da2b77679
